finality-aleph/src/sync/handler/mod.rs

```diff
@@ -511,21 +535,21 @@ where
         justification: J::Unverified,
         maybe_justification: Option<J::Unverified>,
         peer: I,
-    ) -> (Option<BlockIdFor<J>>, Option<<Self as HandlerTypes>::Error>) {
-        let mut maybe_id = None;
+    ) -> (bool, Option<<Self as HandlerTypes>::Error>) {
+        let mut new_highest = false;
 
         for justification in iter::once(justification).chain(maybe_justification) {
-            maybe_id = match self.handle_justification(justification, Some(peer.clone())) {
-                Ok(id) => id,
-                Err(e) => return (maybe_id, Some(e)),
-            };
+            new_highest = match self.handle_justification(justification, Some(peer.clone())) {
+                Ok(new_highest) => new_highest,
+                Err(e) => return (new_highest, Some(e)),
+            } || new_highest;
```

Comment:
Why is it necessary now? Was there a bug?

Reply:
`handle_justification` no longer returns an ID, so it couldn't've stayed as it was. Although I might be misunderstanding your question.

Iirc the underlying modification was made because we are not using the ID anymore, because we don't create a specific task for it, so the ticker only attempts firing, and later gathers the information for the request anyway.